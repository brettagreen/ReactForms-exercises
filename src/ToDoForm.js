import { useState } from 'react';

function ToDoForm({ handleSubmit }) {
    const INITIALSTATE = '';

    const [form, setForm] = useState(INITIALSTATE);

    function handleChange(event) {
        setForm(event.target.value);
    }

    function submitAndClear(event) {
        event.preventDefault();
        handleSubmit(form);
        setForm('');
    }

    return (
        <form onSubmit={submitAndClear}>
            <label htmlFor="task">Enter task:   </label>
            <input data-testid="task-input" id="task" type="text" name="task" value={form} onChange={handleChange} /><br />
            <button>Submit</button>
        </form>
    )
}

export default ToDoForm;