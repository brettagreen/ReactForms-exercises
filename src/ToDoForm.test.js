import { render, fireEvent } from "@testing-library/react";
import ToDoForm from "./ToDoForm";

it("renders without crashing", function() {
  render(<ToDoForm />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<ToDoForm />);
  expect(asFragment()).toMatchSnapshot();
});

it("runs the create function on form submit", function() {
  const createMock = jest.fn();
  const { getByText } = render(<ToDoForm handleSubmit={createMock} />);
  const createButton = getByText("Submit");
  fireEvent.click(createButton);
  expect(createMock).toHaveBeenCalled();
});
