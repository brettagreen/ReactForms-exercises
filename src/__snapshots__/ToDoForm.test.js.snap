// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Enter task:   
    </label>
    <input
      data-testid="task-input"
      id="task"
      name="task"
      type="text"
      value=""
    />
    <br />
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
